dask_bigquery/core.py

```diff
@@ -179,11 +179,7 @@ def make_create_read_session_request(row_filter="", max_stream_count=0):
 
         # Create a read session in order to detect the schema.
         # Read sessions are light weight and will be auto-deleted after 24 hours.
-        session = bqs_client.create_read_session(
-            make_create_read_session_request(
-                row_filter=row_filter, max_stream_count=max_stream_count
-            )
-        )
+        session = bqs_client.create_read_session(make_create_read_session_request())
```

Comment:
A test case for this would be helpful.

Reply:
@j-bennet our existing read test is already a partitioned table, would it be sufficient to just pass `require_partition_filter=True` when we create it?

Reply:
Yes, to make sure you don't get that `InvalidArgument` anywhere.

Reply:
@j-bennet confirmed that the new test fails on `main` and passes now
```
(.venv)  ➜  dask-bigquery git:(main) ✗ pytest dask_bigquery/tests -k test_read_gbq
...
============================= 1 passed, 15 deselected, 2 warnings in 13.88s ==============================

(.venv)  ➜  dask-bigquery git:(main) ✗ gcp 985432d88ef6f2e7f3da8cf5f621a70b4a4080b8
[main 7c201e0] Add require_partition_filter=True to test data creation
 Date: Thu May 11 16:17:11 2023 -0400
 1 file changed, 2 insertions(+), 1 deletion(-)

(.venv)  ➜  dask-bigquery git:(main) ✗ pytest dask_bigquery/tests -k test_read_gbq
...
FAILED dask_bigquery/tests/test_core.py::test_read_gbq - google.api_core.exceptions.InvalidArgument: 400 request failed: Cannot query over table 'model-159019...
============================== 1 failed, 15 deselected, 1 warning in 9.00s ===============================

```

Reply:
@bnaul Now that you did that, you actually need to provide a filter, right? otherwise CI falls apart.

```
FAILED dask_bigquery/tests/test_core.py::test_read_gbq - google.api_core.exceptions.InvalidArgument: 400 request failed: Cannot query over table '***.7900963a0b84469f87881518aab32e0b.table_test' without a filter over column(s) 'timestamp' that can be used for partition elimination
FAILED dask_bigquery/tests/test_core.py::test_read_row_filter - google.api_core.exceptions.InvalidArgument: 400 request failed: Cannot query over table '***.d367d6b552724b1daee1bc88d9b25e49.table_test' without a filter over column(s) 'timestamp' that can be used for partition elimination
FAILED dask_bigquery/tests/test_core.py::test_read_kwargs - google.api_core.exceptions.InvalidArgument: 400 request failed: Cannot query over table '***.d12612b81bb24d869ed0315c3bdad9b5.table_test' without a filter over column(s) 'timestamp' that can be used for partition elimination
FAILED dask_bigquery/tests/test_core.py::test_read_columns - google.api_core.exceptions.InvalidArgument: 400 request failed: Cannot query over table '***.b68dc77b147d4697919ffc066020a44f.table_test' without a filter over column(s) 'timestamp' that can be used for partition elimination
FAILED dask_bigquery/tests/test_core.py::test_max_streams - google.api_core.exceptions.InvalidArgument: 400 request failed: Cannot query over table '***.36f5d0e1feb84d9fb711d3fa130cc100.table_test' without a filter over column(s) 'timestamp' that can be used for partition elimination
FAILED dask_bigquery/tests/test_core.py::test_arrow_options - google.api_core.exceptions.InvalidArgument: 400 request failed: Cannot query over table '***.08e42efc864b4300a9f7379972372d07.table_test' without a filter over column(s) 'timestamp' that can be used for partition elimination
```